### PR TITLE
Fix workspace menu and connect button issues

### LINKED ISSUE #3819: Workspace service link s not working in left nav
The menu navigation on the left hand side doesn’t work. I can expand and contract the menus, and when I select one, I get that entry, but once I’ve selected one, I can’t navigate to any other menu item unless I go to ‘Overview’ first. For example, with the menu as you see in this screenshot, if I click on Overview, then Gitea, I can click all the other items afterwards as much as I like, nothing changes until I go back to Overview, then select another item.

### PR DESCRIPTION
**Pull request by marrobi**
Fixes #3819 #3820 

This pull request fixes two issues in the code:

1. The workspace menu was not working correctly.

2. The connect button was showing even when there was no URI available.

These issues have been resolved by updating the code in the affected files.

### CODE REVIEW

**Comment by marrobi** [comment]
@damoodamoo @jjgriff93 if have any ideas what's caused the workspace menu issues would love to know, but this seems to fix. Thanks,

File: ui/app/src/components/shared/ResourceContextMenu.tsx [hunk 1]
```diff
@@ -142,7 +142,7 @@ export const ResourceContextMenu: React.FunctionComponent<ResourceContextMenuPro
   }
 
   // add 'connect' button if we have a URL to connect to
-  if (props.resource.properties.is_exposed_externally === true) {
+  if (props.resource.properties.is_exposed_externally === true && props.resource.properties.connection_uri) {
```
Comment (jjgriff93) [inline ui/app/src/components/shared/ResourceContextMenu.tsx:145]: 'nit' perhaps wrap the exposed_externally conditional in another that checks for connection_uri

**Review by jjgriff93 — APPROVED** [review]

**Comment by jjgriff93** [comment]
Looks good - and no idea to be honest

**Pull request merged**